subsys/lorawan/lorawan.c

```diff
@@ -472,6 +472,26 @@ int lorawan_set_class(enum lorawan_class dev_class)
 	return 0;
 }
 
+int lorawan_set_channels_mask(uint16_t *channels_mask, uint8_t channels_mask_size)
+{
+	MibRequestConfirm_t mib_req;
+
+	if ((channels_mask == NULL) || (channels_mask_size != CONFIG_LORAWAN_CHANNELS_MASK_SIZE)) {
```

Comment:
If length validation is still desired, it should be done depending on the currently enabled region, not a hardcoded value.

Reply:
Fixed.